### PR TITLE
Computation of angles and areas in hyperbolic geometry

### PR DESCRIPTION
**Pull request by videlec**
Checklist
* [x] Added an entry in `doc/news/`. 
* [x] Added a test for this change.
* [ ] Added new .py files to the documentation in `doc/geometry` or `doc/graphical`.

### CODE REVIEW

**Comment by videlec** [comment]
@saraedum I am not happy with my `boundary_point_is_between` which converts to `RDF`. Could you help? All the rest is hopefully fine (though one might want `area` at a higher level, namely for any convex set).

File: flatsurf/geometry/euclidean.py [hunk 1]
```diff
@@ -151,6 +151,26 @@ def is_cosine_sine_of_rational(cos, sin, scaled=False):
...
+def numerical_cos_to_angle(cos_angle, numerical=False):
+    import math
+
+    angle = math.acos(cos_angle) / (2 * math.pi)  # rat number between 0 and 1/2
+
+    if numerical:
+        return angle
+
+    # fast and dirty way using floating point approximation
+    # (see below for a slow but exact method)
```
Comment (saraedum) [inline flatsurf/geometry/euclidean.py:163]: There's no slow method here.

File: flatsurf/geometry/hyperbolic.py [hunk 2]
```diff
@@ -14266,3 +14408,34 @@ def _lt_(cls, lhs, rhs):
         return geodesic.parametrize(
             lhs.start(), model="euclidean"
         ) < geodesic.parametrize(rhs.start(), model="euclidean")
+
+
+def boundary_point_is_between(p0, p1, q):
```
Comment (saraedum) [inline flatsurf/geometry/hyperbolic.py:14413]: Is this not just the question whether q is in the half space left or right of the geodesic p0-p1?

Comment (videlec) [reply flatsurf/geometry/hyperbolic.py:14413]: It is!

**Comment by videlec** [comment]
I opened https://github.com/flatsurf/exact-real/issues/177 for the problem with macos-11 runner

File: flatsurf/geometry/hyperbolic.py [hunk 3]
```diff
@@ -6174,9 +6242,14 @@ def __contains__(self, point):
...
-                "cannot decide whether this ideal point is contained in the half space yet"
-            )
+            from sage.rings.qqbar import AA
+
+            if AA.has_coerce_map_from(self.parent().base_ring()):
+                x, y = point.coordinates(model="klein", ring=AA)
+            else:
+                raise NotImplementedError(
+                    "cannot decide whether this ideal point is contained in the half space yet"
+                )
```
Comment (saraedum) [inline flatsurf/geometry/hyperbolic.py:6252]: If you don't mind, I'll implement this properly (without `AA`.)

Comment (saraedum) [reply flatsurf/geometry/hyperbolic.py]: Needs some more testing but the implementation should be there now.

**Comment by saraedum** [comment]
@videlec I think I am done here for the moment. Please have a look!

**Comment by videlec** [comment]
Looks good to me. Thanks!

**Pull request merged**